File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        // Brute Force =>TC:O(n^2),SC:O(1)

        // int maxLength=Integer.MIN_VALUE;
        // for(int i=0;i<nums.length;i++){
        // int sum=0;
        // for(int j=i;j<nums.length;j++){
        // sum+=nums[j];
        // maxLength=Math.max(sum,maxLength);
        // }
        // }
        // return maxLength;

        // KADANE'S ALGORITHM(Optimal Soln's)
        int maxLength = Integer.MIN_VALUE;
        int sum = 0;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            maxLength = Math.max(maxLength, sum);
            if (sum < 0)
                sum = 0;
        }

        return maxLength;

    }
}

File: 53-maximum-subarray/maximum-subarray.ts
function maxSubArray(nums: number[]): number {
    let maxSum:number = Math.max(...nums);
    let currSum:number = 0;
    for(let value of nums){
        currSum += value;
        maxSum = Math.max(currSum,maxSum);
        if(currSum<0) currSum=0;
    }
    return maxSum
    
};